### PR TITLE
enhancement(docker source): Renamed docker source to docker_logs

### PR DESCRIPTION
**Pull request by jamtur01**
Signed-off-by: James Turnbull <james@lovedthanlost.net>

### CODE REVIEW

**Comment by jamtur01** [comment]
I might need a touch of help working out how to alias this so that it's less of a breaking change. I'll ping folks tomorrow.

File: src/sources/docker_logs.rs [hunk 1]
```diff
@@ -4,9 +4,10 @@ use crate::{
     event::merge_state::LogEventMergeState,
     event::{self, Event, LogEvent, Value},
     internal_events::{
-        DockerCommunicationError, DockerContainerEventReceived, DockerContainerMetadataFetchFailed,
-        DockerContainerUnwatch, DockerContainerWatch, DockerEventReceived,
-        DockerLoggingDriverUnsupported, DockerTimestampParseFailed,
+        DockerCommunicationError, DockerContainerMetadataFetchFailed,
+        DockerLoggingDriverUnsupported, DockerLogsContainerEventReceived,
```
Comment (fanatid) [inline src/sources/docker_logs.rs:8]: Should we also rename `DockerCommunicationError`, `DockerContainerMetadataFetchFailed`, `DockerLoggingDriverUnsupported`?

**Review by fanatid — APPROVED** [review]

**Comment by bruceg** [comment]
> I might need a touch of help working out how to alias this so that it's less of a breaking change. I'll ping folks tomorrow.

I made an alias for the prometheus sink rename to `prometheus_exporter`. You can see it in #4856 (https://github.com/timberio/vector/blob/37719591828ec599bd3d93a8e1baf2defc2932ec/src/sinks/prometheus/exporter.rs#L121)

**Comment by jamtur01** [comment]
@bruceg Could I get you to throw your eyes over the alias to make sure it looks good? Thanks!

**Comment by bruceg** [comment]
> @bruceg Could I get you to throw your eyes over the alias to make sure it looks good? Thanks!

LGTM :+1:

**Pull request merged**